cdragontoolbox/arenadata.py

```diff
@@ -0,0 +1,121 @@
+from .binfile import BinFile, BinHashBase
+from .rstfile import RstFile
+
+
+class NaiveJsonEncoder(json.JSONEncoder):
+    def default(self, other):
+        if isinstance(other, BinHashBase):
+            if other.s is None:
+                return other.hex()
+            return other.s
+        return other.__dict__
+
+
+class ArenaTransformer:
+    def __init__(self, input_dir):
+        self.input_dir = input_dir
+
+    def build_template(self):
+        """Parse bin data into template data"""
+        map30_file = os.path.join(self.input_dir, "data", "maps", "shipping", "map30", "map30.bin")
+
+        map30 = BinFile(map30_file)
+
+        augments = self.parse_augments(map30)
+
+        return {
+            "augments": augments,
+        }
+
+    def export(self, output, langs=None):
+        """Export Arena data for given languages
+
+        By default (`langs` is `None`), export all available languages.
+        Otherwise, export for given `xx_yy` language codes.
+        """
+
+        stringtable_dir = os.path.join(self.input_dir, "data/menu")
+        stringtable_pattern = "main_??_??.stringtable"
+        stringtable_format = "main_%s.stringtable"
+
+        if langs is None:
+            langs = []
+            for path in glob.glob(os.path.join(stringtable_dir, stringtable_pattern)):
+                # Note: must be adjusted if format changes
+                m = re.search(r'/\w*_(.._..)\.\w*$', path)
+                if m:
+                    langs.append(m.group(1))
+
+        os.makedirs(output, exist_ok=True)
+
+        template = self.build_template()
+        for lang in langs:
+            instance = copy.deepcopy(template)
+            replacements = RstFile(os.path.join(stringtable_dir, stringtable_format % lang))
+
+            def replace_in_data(entry):
+                for key in ("name", "desc", "tooltip"):
+                    if key in entry and entry[key] in replacements:
+                        entry[key] = replacements[entry[key]]
+
+            for augment in instance["augments"]:
+                replace_in_data(augment)
+
+            with open(os.path.join(output, f"{lang}.json"), "w", encoding="utf-8") as f:
+                json.dump(instance, f, cls=NaiveJsonEncoder, indent=4, sort_keys=True)
+
+    def parse_augments(self, map30):
+        """Returns a list of augments"""
+        augment_entries = [x for x in map30.entries if x.type == 0x6DFAB860]
+        spellobject_entries = {x.path: x for x in map30.entries if x.type == "SpellObject"}
+
+        augments = []
+        for augment in augment_entries:
+
+            augment_datavalues = {}
+            augment_spellobject = augment.getv(0x1418F849)
+            if augment_spellobject:
+                augment_spell = spellobject_entries[augment_spellobject].getv('mSpell')
+                for datavalue in augment_spell.getv('mDataValues', []):
+                    augment_datavalues[datavalue.getv("mName")] = datavalue.getv("mValues", [None])[0]
```

Comment:
Internally, `mValues` is an array of floats and the actual default value is `0`, which you should use instead of `None`.
(There is currently one augment in this case.)